### PR TITLE
Weba/merge ngemm

### PR DESCRIPTION
**Pull request by baowenlei**
**Description**: Add Tensorize GEMM support including AVX2/AVX512.

**Motivation and Context**
- Our approach is described in the paper here https://arxiv.org/abs/1910.00178

### CODE REVIEW

File: onnxruntime/core/providers/nuphar/nuphar_execution_provider.cc [hunk 1]
```diff
@@ -257,6 +263,9 @@ NupharExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_vie
     for (const auto& node_idx : result->sub_graph->nodes) {
       const auto& node = graph_viewer.GetNode(node_idx);
 
+      if (node->Name() == "Times73164MatMulInteger_14")
+        std::cout << "name: const inti" << node->Name() << std::endl;
```
Comment (KeDengMS) [inline onnxruntime/core/providers/nuphar/nuphar_execution_provider.cc]: remove?

File: onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc [hunk 2]
```diff
@@ -6,9 +6,29 @@
...
 
+std::unique_ptr<CodeGenTarget> CodeGenTarget_AVX() {
+  // Make it conservative by enabling basic avx support for sandy-bridge cpus
+  // Also note that AVX-1 has 256-bit vector support only for float. For int,
+  // AVX-1 still uses 128-vector. Let's be conservative as well by choosing
+  // 128 as max_vector_bits. We can change the value to 256 if later we think
+  // 256 is more appropriate. Moreover, the value only affects performance for avx-1,
+  // which we care about less. I couldn't find the point in Intel's reference
+  // where the number of vector units is mentioned explicitly, but got it
+  // from some forum by google search.
```
Comment (KeDengMS) [inline onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc]: seems unnecessary comments

File: onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc [hunk 3]
```diff
@@ -6,9 +6,29 @@
...
+  // TODO: we should refine vector_unit_num if it turned out not to be 2.
+  // The current number is based on:
+  // https://en.wikipedia.org/wiki/Sandy_Bridge
+  // "Improved 3 integer ALU, 2 vector ALU and 2 AGU per core"
+  // and
+  // https://indico.cern.ch/event/625333/contributions/2587012/attachments/1492809/2330113/VectorParallelism.pdf
+  // "Example: Intel Xeon E5-2670 v2 �Ivy Bridge� �10 cores, each with 256-bit AVX vector unit "
```
Comment (KeDengMS) [inline onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc]: >2 �Ivy Bridge� �10 c [](start = 35, length = 20)

please tidy it up to have only standard ascii chars.

File: onnxruntime/core/providers/nuphar/nuphar_execution_provider.cc (hunk 1 above)
Comment (baowenlei) [reply onnxruntime/core/providers/nuphar/nuphar_execution_provider.cc]: Thanks for catch that.

File: onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc (hunk 3 above)
Comment (baowenlei) [reply onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc]: Removed.

File: onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc (hunk 2 above)
Comment (baowenlei) [reply onnxruntime/core/providers/nuphar/compiler/x86/x86_target_info.cc]: Removed

**Pull request merged**